Scripts/Runtime/Entities/TaskSystem/Job/Wrapper/CancelCompleteDataStreamAccessWrapper.cs

```diff
@@ -0,0 +1,27 @@
+using Anvil.Unity.DOTS.Jobs;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    internal class CancelCompleteDataStreamAccessWrapper : AbstractAccessWrapper
+    {
+        public CancelCompleteDataStream CancelCompleteDataStream { get; }
+
+        public CancelCompleteDataStreamAccessWrapper(CancelCompleteDataStream cancelCompleteDataStream,
+                                                     AccessType accessType,
+                                                     AbstractJobConfig.Usage usage) : base(accessType, usage)
+        {
+            CancelCompleteDataStream = cancelCompleteDataStream;
+        }
+
+        public override JobHandle Acquire()
+        {
+            return CancelCompleteDataStream.AccessController.AcquireAsync(AccessType);
+        }
+
+        public override void Release(JobHandle releaseAccessDependency)
+        {
+            CancelCompleteDataStream.AccessController.ReleaseAsync(releaseAccessDependency);
+        }
```

Comment:
How complex do these get in other variants?
Could the `AbstractAccessWrapper` just taken in the `AccessController` instance and do the acquire/release rather than creating a specialty type for each type of stream.

If it's not always access controllers should we have an interface that describes something as access controlled?
Ex: `IAccessControlled` with `AcquireAsync` and `ReleaseAsync` then you can pass anything that satisfies that contract without needing specialty wrappers.

Reply:
I will double check on this. Good point.